console/console-init/ui/src/graphql-module/queries/iot_device.ts

```diff
@@ -101,16 +101,45 @@ const SORT_RETURN_ALL_DEVICES_FOR_IOT_PROJECT = (sortBy?: ISortBy) => {
   return orderBy;
 };
 
+const concatAND = (filter: string) => {
+  if (filter?.length > 0) return " AND ";
```

Comment:
if condition needs to be modified. since it is a string we should replace the condition with `if(filter?.trim()!==""`